src/act.drive.cpp

```diff
@@ -1268,19 +1268,23 @@ ACMD(do_driveby)
     send_to_char("How do you expect to driveby from a stationary vehicle?\r\n", ch);
     return;
   }
-
+ 
   if ((wielded = GET_EQ(ch, WEAR_WIELD))) {
-    if (!(GET_OBJ_VAL(wielded, 4) == SKILL_PISTOLS || GET_OBJ_VAL(wielded, 4) == SKILL_SMG)) {
-      send_to_char(ch, "You can only do a driveby with a pistol or SMG.\r\n");
+    if (GET_OBJ_TYPE(wielded) != ITEM_WEAPON) {
+      send_to_char(ch, "You have to be wielding a weapon!"
+      return;
+    }
+    if ((IS_OBJ_STAT(wielded, ITEM_EXTRA_TWOHANDS) || (!WEAPON_IS_GUN(wielded) && ch->in_veh->type = !VEH_BIKE)))  {
+      send_to_char(ch, "You can only do a driveby with a one handed weapon in a vehicle and a melee weapon while on a bike!\r\n");
       return;
     }
     if (!IS_NPC(vict) && !PRF_FLAGGED(ch, PRF_PKER) && !PRF_FLAGGED(vict, PRF_PKER)) {
       send_to_char(ch, "You have to be flagged PK to attack another player.\r\n");
       return;
     }
 
-    send_to_char(ch, "You point %s towards %s and open fire!\r\n", GET_OBJ_NAME(wielded), GET_NAME(vict));
-    snprintf(buf, sizeof(buf), "%s aims %s towards %s and opens fire!\r\n", GET_NAME(ch), GET_OBJ_NAME(wielded), GET_NAME(vict));
+    send_to_char(ch, "You swing %s towards %s and attack!\r\n", GET_OBJ_NAME(wielded), GET_NAME(vict));
```

Comment:
Instead of rewriting all driveby attack messages, print a different attack message depending on the weapon type used.